packages/libs/error-stack/src/report.rs

```diff
@@ -617,10 +619,52 @@ impl<Context> fmt::Debug for Report<Context> {
     }
 }
 
+#[cfg(feature = "std")]
+impl<Context> std::process::Termination for Report<Context> {
+    fn report(self) -> ExitCode {
+        #[cfg(not(nightly))]
+        return ExitCode::FAILURE;
+
+        #[cfg(nightly)]
+        self.request_ref::<ExitCode>()
+            .next()
+            .copied()
+            .unwrap_or(ExitCode::FAILURE)
+    }
+}
+
 pub struct ReportImpl {
     pub(super) frame: Frame,
     #[cfg(all(nightly, feature = "std"))]
     backtrace: Option<Backtrace>,
     #[cfg(feature = "spantrace")]
     span_trace: Option<SpanTrace>,
 }
+#[cfg(test)]
+mod test {
+    use std::process::{ExitCode, Termination};
+
+    use crate::{Context, Report};
+
+    #[derive(Debug)]
+    struct CustomError;
+
+    impl Context for CustomError {}
+
+    impl std::fmt::Display for CustomError {
+        fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+            f.write_str("Custom Error")
+        }
+    }
+
+    #[test]
+    fn test_exit_code() {
+        let report = Report::new(CustomError)
+            .attach(ExitCode::from(100))
+            .attach_printable("This error has an exit code of 100!");
+
+        // _ to get rid of clippy warning.
+        let _exit = report.report();
+        assert!(matches!(ExitCode::from(100), _exit));
+    }
+}
```

Comment:
I didn't know that `ExitCode` does not implement `PartialEq`. Without the ability to compare the exit code the test isn't very useful, so let's remove the test again. I'm sorry for the inconvenience! (At least we have the example)

```suggestion
```

Reply:
All good!